### PR TITLE
New Component IOTA

### PR DESCRIPTION
**Pull request by Fabryprog**

### CODE REVIEW

**Comment by Fabryprog** [comment]
sorry I forgot to send the latest version of the adoc file. I will send it as soon as possible

File: parent/pom.xml [hunk 1]
```diff
@@ -399,6 +399,7 @@
     <jmh-version>1.21</jmh-version>
     <jodatime-bundle-version>1.6.2</jodatime-bundle-version>
     <jodatime2-bundle-version>2.10</jodatime2-bundle-version>
+    <jota-bundle-version>0.9.10</jota-bundle-version>
```
Comment (oscerd) [inline parent/pom.xml]: Is Jota Java library a bundle? Not sure.

File: components/camel-iota/pom.xml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		<camel.osgi.export.service>org.apache.camel.spi.ComponentResolver;component=iota</camel.osgi.export.service>
+	</properties>
+
+	<repositories>
+		<repository>
+		    <id>spring-plugin</id>
+		    <url>http://repo.spring.io/plugins-release</url>
+		</repository>
+	</repositories>
+
```
[the remark this answers is not in the document]
Comment (Fabryprog) [reply components/camel-iota/pom.xml]: Hello, official maven repo contains an old version 0.9.7. Into spring.io repo you can find latest version!

File: parent/pom.xml (hunk 1 above)
Comment (Fabryprog) [reply parent/pom.xml]: No it's a lib sorry! It is a mistake

File: components/camel-iota/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (oscerd) [reply components/camel-iota/pom.xml]: We can't rely only on Spring.io repository. For the moment it's fine, but once we have 0.9.10 on central, this repo must be removed.

**Comment by Fabryprog** [comment]
@oscerd my latest commit fix sourcecheck  and remove useless code!

**Comment by oscerd** [comment]
By the way I completely cannot find this stuff in Maven central. Do they plan to release this stuff? Or is it a pre-release?

**Comment by Fabryprog** [comment]
Omg it is right! 0.9.7 is also on spring lib repo (https://mvnrepository.com/artifact/org.iota/jota?repo=springio-libs-release)

So.. i am writing to develop team and i will write you the feedback.

File: components/camel-iota/src/main/java/org/apache/camel/component/iota/IOTAConstants.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    public static final String SEED_HEADER = "CamelIOTASeed";
+    public static final String VALUE_HEADER = "CamelIOTAValue";
+    public static final String TO_ADDRESS_HEADER = "CamelIOTAToAddress";
+    public static final String ADDRESS_INDEX_HEADER = "CamelIOTAAddressIndex";
+
+    protected static final int MIN_WEIGHT_MAGNITUDE = 14;
+    protected static final int DEPTH = 9;
+
+    protected static final int TAG_LENGTH = Constants.TAG_LENGTH;
```
Comment (onderson) [inline components/camel-iota/src/main/java/org/apache/camel/component/iota/IOTAConstants.java]: do we really need this?

Comment (Fabryprog) [reply components/camel-iota/src/main/java/org/apache/camel/component/iota/IOTAConstants.java]: I think so

**Comment by oscerd** [comment]
I'll merge this tomorrow.

**Comment by oscerd** [comment]
I'd like to have the opinion of @davsclaus too about library from outside of cental by the way

**Comment by Fabryprog** [comment]
@oscerd my last commit add a new operation to retrieve information from tangle!

**Comment by davsclaus** [comment]
Yeah we would like to avoid 3rd party maven repos. So if you can use http rest then that is good too.
An alternative sometimes is to have SMX team release an OSGi bundle of a library and release it to maven central, then the component can use that JAR instead.

**Comment by Fabryprog** [comment]
Hello @davsclaus and @oscerd i commit latest component version.

i deleted java dependency and implement the 3 request on the tangle using http rest

 - FIND_TRANSACTION_BY_ADDRESS
 - FIND_TRANSACTION_BY_TAG
 - FIND_TRANSACTION_DATA

Now i am starting to write "SEND_TRANSACTION" operation.

So.. you can merge this pull request or wait for new code.

P.S. maybe it's better to do after a new PR cointains the new implementation!

King regards

**Comment by oscerd** [comment]
Lets wait for the final implementation.

**Comment by Fabryprog** [comment]
Hello everyone.

I am trying to write the iota transaction's business logic into component. I am finishing implementation but there are some classes copied from official java lib and some classes to convert into ternary code (iota is based to trit). This classes aren't complex but these so is boring code!

So i doubt to use this approch because it can be a very future problem (imaging iota update protocol)

What do I do?

Can i clpse this pull request without merge or include the official java lib using an external maven repo?

**Comment by oscerd** [comment]
I believe we had problems in the past with external repository. But lets wait for the others @davsclaus @lburgazzoli @nicolaferraro @onderson @dmvolod

**Comment by dmvolod** [comment]
@Fabryprog , yes it's possible workaround, but in this case a separate from "org.iota" groupId will be using.

**Comment by onderson** [comment]
@Fabryprog , good idea. You will have the burden of maintaining that library. If we have such practices +1,  otherwise -1 = 0 for now :-)

**Comment by Fabryprog** [comment]
@dmvolod package refactoring can be a nightmare. There is an alternative solution?

**Comment by oscerd** [comment]
No other solutions. This can create problem. It looks troublesome to have this component in Camel.

**Comment by dmvolod** [comment]
You can ping component dev again, for publishing lib to the official maven repo. For example, Apache Thrift team usually delay with publishing.

**Comment by Fabryprog** [comment]
@dmvolod I spent a couple of days attached to the legs of the team without result... :-(

**Comment by Fabryprog** [comment]
@oscerd maybe it's better to dedicate yourself to something else

**Comment by oscerd** [comment]
If you just want PMC saying yes to everything, this is not possible.

**Comment by Fabryprog** [comment]
@oscerd You misunderstood me. I would used the plural :-) sorry for the misunderstanding but the Phone is switch off...

Sorry man!!!!

**Comment by Fabryprog** [comment]
guys for now I close the issue without any merge. 

when the team put the lib on the central repository, I will try to rewrite the component.

thank you for your time and support

step and close

**Pull request closed without merging**